### ISSUE TITLE
Should not be accessible to the group unless group member tries

### ISSUE THREAD

**Issue by Silver-IT** [okTurtles/group-income#1716]
### Problem

Steps to reproduce
- user1 creates a group
- user2 joins the group with the invitation link and close browser
- user3 joins the group with the invitation link
- user3 creates a proposal to remove user2
- user1 aceepts the proposal
- user2 logs in and access the group even though he is not the group member

This error happens ONLY when the group member is kicked by proposal.

### Solution

Find the reason and fix it

**Comment by taoeffect** [comment]
@Silver-IT This is probably not an issue on `e2e-protocol`

**Comment by Silver-IT** [comment]
@taoeffect, I have tested it in the `master` branch, and it doesn't happen.

**Comment by Silver-IT** [comment]
@taoeffect, you are right. Having investigated this issue more in detail, I have found that the reason is the workflow of login process.
At the begining of the `gi.actions/group/join` function, we can see `sbp('okTurtles.data/set', 'JOINING_GROUP', true)`.
This doesn't make sense, because we could call `gi.actions/group/join` function when login.

And the reason why this issue didn't happen last week is because the `groupsJoined` was empty array when I tested.
```
const groupsJoined = difference(contractLoginState.groupIds, ourLoginState.groupIds)
```
This happens when each group member uses the one browser and the vuex state never be cleared.

**Comment by Silver-IT** [comment]
In `master` branch, it can be fixed with the following changes.
![image](https://github.com/okTurtles/group-income/assets/57976479/1a5e5c9d-5535-4d65-9e1a-981611fe2b03)

However, I will fix this issue in [this branch](https://github.com/okTurtles/group-income/tree/1716-should-not-be-accessible-to-the-group-unless-group-member-tries).

**Comment by Silver-IT** [comment]
@corrideat. Since this issue is more related to the e2e part, I agree on that you are going to work on this.
Having investigated, I don't think we need the `skipInviteAccept` optional parameter in `gi.actions/group/join` function.
That's because we can use `params.originatingContractID` instead of `skipInviteAccept`.
I think the issue #1765 could be related too.

cc: @taoeffect 

EDITED: I will remove this `skipInviteAccept` paremeter while working on this issue. https://github.com/okTurtles/group-income/issues/1766#issuecomment-1763548343

**Comment by Silver-IT** [comment]
@corrideat, Please check this [chat history](https://okturtles.slack.com/archives/C02QF71TK1Q/p1698054340239609) and files attached. It is the solution of this issue.
I will handle this issue since it's not related to the low-level code of e2e part.

cc: @taoeffect

**Comment by taoeffect** [comment]
@corrideat this is still an issue in `e2e-protocol-ricardo` when I replace this last step:

> user2 logs in and access the group even though he is not the group member

With:

> user2 logs in a private window and  has access to the group even though he is not the group member

**Comment by taoeffect** [comment]
Fixed in d9de01802c5e2e3c3b0373dea0eda0d4a998e668
